### PR TITLE
Fixes for xmc4700-relax board

### PR DESCRIPTION
**Pull request by trns1997**
## Summary
1. Added SPI interface according to the evaluation board doc (https://www.infineon.com/dgdl/Infineon-Board_User_Manual_XMC4700_XMC4800_Relax_Kit_Series-UserManual-v01_04-EN.pdf?fileId=5546d46250cc1fdf01513f8e052d07fc) USIC2 Channel 0.
2. Fix userled driver so that LED1 and LED2 are correctly routed.

## Impact
N/A

## Testing
N/A

### CODE REVIEW

**Comment by xiaoxiang781216** [comment]
@trns1997 let's squash your last two style fix patch to the first two.

**Comment by trns1997** [comment]
@xiaoxiang781216 noted, but aren't they automatically squashed when merged ?

**Comment by xiaoxiang781216** [comment]
> @xiaoxiang781216 noted, but aren't they automatically squashed when merged ?

No, the action is "Rebase and merge" to keep the origin patch layout.

**Comment by acassis** [comment]
> @acassis noted I will fix the styling errors. Just a side question I created a small example that uses the SPI to communicate with a LSM6DSR imu. I created it in the nuttx-apps project. Do you think its worth making a PR over there?

@trns1997 yes, all examples help new users to test the hardware. Also note that new sensor drivers can use uORB protocol. So, you can adapt the existing driver to use uORB protocol with app testing sensortest or add your example as a testing example.

**Comment by trns1997** [comment]
> > @acassis noted I will fix the styling errors. Just a side question I created a small example that uses the SPI to communicate with a LSM6DSR imu. I created it in the nuttx-apps project. Do you think its worth making a PR over there?
> 
> @trns1997 yes, all examples help new users to test the hardware. Also note that new sensor drivers can use uORB protocol. So, you can adapt the existing driver to use uORB protocol with app testing sensortest or add your example as a testing example.

I think I will look into this later, I am not familiar with the uORB protocol but i can see that there are examples that use it. For now I directly call the macros from `nuttx/spi/spi.h` in my application to test the SPI communication with the sensor. But I think I will dev a sensor driver that uses the registered spi driver which seems to be the general approach here. Sorry I am fairly new to nuttx soo there is a lot to take in.

**Comment by acassis** [comment]
@trns1997 sure!

Please remove the Draft to let me merge it

File: arch/arm/src/xmc4/xmc4_spi.c [hunk 1]
```diff
@@ -234,7 +233,7 @@ static inline struct xmc4_spidev_s *spi_device(struct xmc4_spics_s *spics);
 #ifdef CONFIG_DEBUG_SPI_INFO
 static void     spi_dumpregs(struct xmc4_spidev_s *spi, const char *msg);
 #else
-#  define       spi_dumpregs(spi, msg)
+# define        spi_dumpregs(spi, msg)
```
Comment (xiaoxiang781216) [inline arch/arm/src/xmc4/xmc4_spi.c:236]: revert the change

File: arch/arm/src/xmc4/xmc4_spi.c [hunk 2]
```diff
@@ -222,7 +221,7 @@ static bool     spi_checkreg(struct xmc4_spidev_s *spi, bool wr,
                              uint32_t value, uint32_t address);
 
 #else
-#  define       spi_checkreg(spi, wr, value, address)  (false)
+# define        spi_checkreg(spi, wr, value, address)  (false)
```
Comment (xiaoxiang781216) [inline arch/arm/src/xmc4/xmc4_spi.c:224]: revert the change

File: boards/arm/xmc4/xmc4700-relax/src/xmc4_bringup.c [hunk 3]
```diff
@@ -26,9 +26,15 @@
 
 #include <debug.h>
 #include <errno.h>
-#include <nuttx/spi/spi_transfer.h>
 #include <sys/types.h>
 
+#ifdef CONFIG_XMC4_USCI_SPI
+  #include <nuttx/spi/spi_transfer.h>
```
Comment (xiaoxiang781216) [inline boards/arm/xmc4/xmc4700-relax/src/xmc4_bringup.c:32]:
```
#  include <nuttx/spi/spi_transfer.h>
```

File: boards/arm/xmc4/xmc4700-relax/src/xmc4_bringup.c [hunk 4]
```diff
@@ -26,9 +26,15 @@
...
+#endif
+#ifdef CONFIG_USERLED
+  #include <nuttx/leds/userled.h>
```
Comment (xiaoxiang781216) [inline boards/arm/xmc4/xmc4700-relax/src/xmc4_bringup.c:35]:
```
#  include <nuttx/leds/userled.h>
```

Comment (acassis) [reply boards/arm/xmc4/xmc4700-relax/src/xmc4_bringup.c:35]: @trns1997 as @trns1997 the correct is "#  " instead "  #".
First time I saw it I was surprised because I didn't know that the compiler was able to recognize space after # as valid directive.

Comment (trns1997) [reply boards/arm/xmc4/xmc4700-relax/src/xmc4_bringup.c]: Alright noted, sorry just my usual coding habit. It is more readable your way though I guess I might take it up :)

Comment (xiaoxiang781216) [reply boards/arm/xmc4/xmc4700-relax/src/xmc4_bringup.c]: Don't worry, @trns1997 but again, it's better to amend the style change to the first patch.

**Review by acassis — APPROVED** [review]

File: boards/arm/xmc4/xmc4700-relax/src/xmc4_bringup.c (hunk 4 above)
Comment (acassis) [reply boards/arm/xmc4/xmc4700-relax/src/xmc4_bringup.c]: Sorry, I missed this comment, in fact, he left some minor issues in the first commit "# define" instead of "#  define".
@trns1997 when you submit a new PR to XMC4700, please remember to fix those modificações you did in the source code
